app/src/main/java/fr/free/nrw/commons/filepicker/UploadableFile.java

```diff
@@ -124,10 +124,10 @@ private DateTimeWithSource getFileCreatedDateFromCP(Context context) {
     private DateTimeWithSource getDateTimeFromExif() {
         try {
             ExifInterface exif = new ExifInterface(file.getAbsolutePath());
-            @SuppressLint("RestrictedApi") long dateTime = exif.getDateTime();
-            if (dateTime != -1) {
-                Date date = new Date(dateTime);
-                return new DateTimeWithSource(date, DateTimeWithSource.EXIF_SOURCE);
+                @SuppressLint("RestrictedApi") Long dateTime = exif.getDateTime();
```

Comment:
Great! Thanks for finding this!
Last thing, would you mind fixing the indentation?

Reply:
Fixed.